src/promptflow/promptflow/_sdk/entities/_flow.py

```diff
@@ -105,3 +115,66 @@ def _get_local_connections(cls, executable):
                 # ignore when connection not found since it can be configured with env var.
                 raise Exception(f"Connection {n!r} required for flow {executable.name!r} is not found.")
         return result
+
+
+class ProtectedFlow(Flow, SchemaValidatableMixin):
+    def __init__(
+        self,
+        code: str,
+        **kwargs,
+    ):
+        super().__init__(code=code, **kwargs)
+
+        self._flow_dir, self._dag_file_name = self._get_flow_definition(self.code)
+
+    @property
```

Comment:
Seems some properties can be put in Flow class?

Reply:
Hide all of them for now to avoid impacting public interface

Reply:
Explained in docstring